activesupport/lib/active_support/message_encryptor.rb

```diff
@@ -86,14 +87,18 @@ def initialize(secret, *signature_key_or_options)
 
     # Encrypt and sign a message. We need to sign the message in order to avoid
     # padding attacks. Reference: http://www.limited-entropy.com/padding-oracle-attacks.
-    def encrypt_and_sign(value)
+    def encrypt_and_sign(value, expires_at: nil, expires_in: nil, purpose: nil)
+      if expires_at || expires_in || purpose
+        value = ActiveSupport::Messages::Metadata.wrap(value, expires_at: expires_at, expires_in: expires_in, purpose: purpose)
+      end
       verifier.generate(_encrypt(value))
     end
 
     # Decrypt and verify a message. We need to verify the message in order to
     # avoid padding attacks. Reference: http://www.limited-entropy.com/padding-oracle-attacks.
-    def decrypt_and_verify(value)
-      _decrypt(verifier.verify(value))
+    def decrypt_and_verify(value, purpose: nil)
+      message = _decrypt(verifier.verify(value))
+      ActiveSupport::Messages::Metadata.verify(message, purpose)
```

Comment:
We're already in the `ActiveSupport` namespace here. So I think we can shorten this to `Messages::Metadata`. Same above.